### PR TITLE
[R20-85] - WIP Primary nav

### PR DESCRIPTION
**Pull request by ChrisCampbellCodes**
**Issue**: https://digital-vic.atlassian.net/browse/R20-85

### What I did:
- Added new `rpl-primary-nav` component
- Refactored 'expandable collection' composable to work off IDs rather than index numbers so that it can be used for multi level content, e.g. no colliding indexs of 0
- Removed early POC `nav-primary` component

## Remaining to do:
- Fix mega menu / search dissapearing before nav closing animation has finished
- Investigate / fix tabbing order in mega nav columns
- Add functionality for nav to show / hide based on page scroll direction
- Add sliding animation to mobile mega nav when changing between columns

### CODE REVIEW

**Comment by lambry** [comment]
@ChrisCampbellCodes It's looking good! 

One thing occurs to me, as the menus aren't nested we lose some default functionality in regards to accessibility, i.e. tabbing into and out of menus.

**Comment by ChrisCampbellCodes** [comment]
> @ChrisCampbellCodes It's looking good!
> 
> One thing occurs to me, as the menus aren't nested we lose some default functionality in regards to accessibility, i.e. tabbing into and out of menus.

ahhh yep that's a good point!

File: examples/nuxt-app/test/features/site/theme.feature [hunk 1]
```diff
@@ -8,13 +8,13 @@ Feature: Site theme
     Given the endpoint "/api/tide/site" with query "?id=8888" returns fixture "/site/reference" with status 200
     And the endpoint "/api/tide/page" with query "?path=/&site=8888" returns fixture "/landingpage/home" with status 200
     Given I visit the page "/"
-    Then the site header background color should be "rgb(128, 0, 128)"
+    # Then the site header background color should be "rgb(128, 0, 128)"
```
Comment (dylankelly) [inline examples/nuxt-app/test/features/site/theme.feature:11]: This should still work though yeah? Is it just the selector is wrong?

File: examples/nuxt-app/test/features/site/theme.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 
-Then('the site header background color should be {string}', (colour: string) => {
-  cy.get(`.rpl-nav-primary`).should('have.css', 'background-color', colour)
-})
-
+// TODO: Uncomment this test once the new primary nav component is ready
+// Then(
+//   'the site header background color should be {string}',
+//   (colour: string) => {
+//     cy.get(`.rpl-primary-nav`).should('have.css', 'background-color', colour)
```
Comment (dylankelly) [inline examples/nuxt-app/test/features/site/theme.ts:7]: Can we reinstate this step definition and update the `cy.get` selector?

File: packages/ripple-ui-core/src/components/layout/layout.stories.mdx [hunk 3]
```diff
@@ -175,19 +188,31 @@ export const PageLayoutNavigationTemplate = (args) => ({
...
   setup() {
     const img = svgPlaceholder({ width: 1200, height: 800, bgColor: '#e76', fgColor: '#a31' })
-    return { args, RplNavCard, img }
+    return { args, primaryNavItems, RplNavCard, img }
   },
   template: `
     <RplLayout v-bind="args">
       <template #primaryNav>
-        <RplNavPrimary></RplNavPrimary>
+        <RplPrimaryNav
```
Comment (dylankelly) [inline packages/ripple-ui-core/src/components/layout/layout.stories.mdx:203]: 👍

File: packages/ripple-ui-core/src/components/layout/layout.stories.mdx [hunk 4]
```diff
@@ -121,7 +122,19 @@ export const PageLayoutContentTemplate = (args) => ({
         />
       </template>
       <template #primaryNav>
-        <RplNavPrimary></RplNavPrimary>
+        <RplPrimaryNav
```
Comment (dylankelly) [inline packages/ripple-ui-core/src/components/layout/layout.stories.mdx:125]: @ChrisCampbellCodes can we remove the margin on the layout header div here https://github.com/dpc-sdp/ripple-framework/blob/develop/packages/ripple-ui-core/src/components/layout/layout.css#L42 in this PR? As we discussed, its better for the primary nav component to look after it.

**Review by dylankelly — CHANGES_REQUESTED** [review]
I can help get that test working, yell out if you want a hand

**Review by dylankelly — APPROVED** [review]

**Comment by dylankelly** [comment]
@ChrisCampbellCodes Tests passing looking goooood! Can we merge it?

**Review by lambry — APPROVED** [review]

**Comment by lambry** [comment]
Hey @Chris, one minor thing, I noticed the quick exit button isn't on desktop.

**Pull request merged**